services/src/tasks/in_memory.rs

```diff
@@ -0,0 +1,150 @@
+use super::{
+    RunningTaskStatusInfo, Task, TaskContext, TaskDb, TaskError, TaskFilter, TaskId,
+    TaskListOptions, TaskStatus, TaskStatusInfo, TaskStatusWithId,
+};
+use crate::{contexts::Db, error::Result};
+use futures::StreamExt;
+use geoengine_datatypes::util::Identifier;
+use std::{
+    collections::{HashMap, VecDeque},
+    sync::Arc,
+};
+use tokio::{
+    sync::{RwLock, RwLockWriteGuard},
+    task::JoinHandle,
+};
+
+/// An in-memory implementation of the `TaskDb` trait.
+#[derive(Default, Clone)]
+pub struct InMemoryTaskDb {
+    handles: Db<HashMap<TaskId, JoinHandle<()>>>,
+    tasks_by_id: Db<HashMap<TaskId, Db<TaskStatus>>>,
+    task_list: Db<VecDeque<TaskUptableStatusWithTaskId>>,
+}
+
+impl InMemoryTaskDb {
+    async fn write_lock_all(&self) -> WriteLockAll {
+        let (handles, tasks_by_id, task_list) = tokio::join!(
+            self.handles.write(),
+            self.tasks_by_id.write(),
+            self.task_list.write()
+        );
+        WriteLockAll {
+            handles,
+            tasks_by_id,
+            task_list,
+        }
+    }
+}
+
+struct TaskUptableStatusWithTaskId {
```

Comment:
what is a TaskUptable?

Reply:
sql'scher Versprecher
[dbd850a](https://github.com/geo-engine/geoengine/pull/555/commits/dbd850a1883a97525514005abd31d62db604d8e8)